dev/checkstyle-suppressions.xml

```diff
@@ -30,6 +30,8 @@
 <suppressions>
     <suppress checks=".*"
               files="core/src/main/java/org/apache/spark/util/collection/TimSort.java"/>
+    <suppress checks="NoFinalizer"
+              files="core/src/main/java/org/apache/spark/io/NioBufferedFileInputStream.java"/>
```

Comment:
Ah, I thought we could disable it only for some lines. I am fine with this if it is only the way.